src/device/pf_pdport.c

```diff
@@ -366,30 +487,169 @@ static void pf_pdport_run_peer_check (
+            "Wanted peer port name: \"%.*s\"\n",
+            __LINE__,
+            (int)lldp_port_id.len,
+            lldp_port_id.string,
+            p_wanted_peer->length_peer_port_name,
+            p_wanted_peer->peer_port_name);
+
+         (void)pf_diag_std_add (
+            net,
+            &diag_source,
+            PNET_DIAG_CH_PROP_TYPE_UNSPECIFIED,
+            PNET_DIAG_CH_PROP_MAINT_FAULT,
+            PF_WRT_ERROR_REMOTE_MISMATCH,
+            PF_WRT_ERROR_PEER_PORTNAME_MISMATCH,
+            0,
+            0);
+      }
+      else
+      {
+         LOG_DEBUG (
+            PNET_LOG,
+            "PDPORT(%d): Peer port name is correct: \"%.*s\". "
+            "Remove diagnosis.\n",
+            __LINE__,
+            (int)lldp_port_id.len,
+            lldp_port_id.string);
+         (void)pf_diag_std_remove (
+            net,
+            &diag_source,
+            PF_WRT_ERROR_REMOTE_MISMATCH,
+            PF_WRT_ERROR_PEER_PORTNAME_MISMATCH);
+      }
+   }
+}
 
-      (void)peer_portname_is_correct; /* TODO use in the decision making */
+/**
+ * @internal
+ * Run peer check
+ *
+ * Compare llpd peer information with configured pdport peer.
+ * Trigger an diagnosis if unexpected peer data is connected.
+ * Remove diagnosis if expected peer data is received.
+ *
+ * @param net              InOut: The p-net stack instance
+ * @param loc_port_num     In:    Local port number.
+ *                                Valid range: 1 .. N, where N is the total
+ *                                number of local ports used by p-net stack.
+ */
+static void pf_pdport_run_peer_check (pnet_t * net, int loc_port_num)
+{
+   pf_port_t * p_port_data = pf_port_get_state (net, loc_port_num);
+
+   if (p_port_data->pdport.check.active == true)
+   {
+      pf_pdport_check_peer_port_name (net, loc_port_num);
+      pf_pdport_check_peer_station_name (net, loc_port_num);
    }
    else
    {
       LOG_DEBUG (
          PNET_LOG,
-         "PDPORT(%d): We do not check peer name and portID, so no port change "
+         "PDPORT(%d): We do not check peer name or portID, so no port change "
          "alarm is sent.\n",
          __LINE__);
    }
 }
 
+static void pf_pdport_peer_lldp_timeout_alarm (pnet_t * net, int loc_port_num)
+{
+   pnet_diag_source_t diag_source = {0};
+
+   pf_pdport_init_diag_source (&diag_source, loc_port_num);
+
+   LOG_DEBUG (
+      PNET_LOG,
+      "PDPORT(%d): Sending no peer detected alarm port %u",
+      __LINE__,
+      loc_port_num);
+
+   (void)pf_diag_std_add (
+      net,
+      &diag_source,
+      PNET_DIAG_CH_PROP_TYPE_UNSPECIFIED,
+      PNET_DIAG_CH_PROP_MAINT_FAULT,
+      PF_WRT_ERROR_REMOTE_MISMATCH,
+      PF_WRT_ERROR_NO_PEER_DETECTED,
+      0,
+      0);
+}
+
+void pf_pdport_periodic (pnet_t * net)
+{
+   int loc_port_num;
+   pf_port_t * p_port_data;
+
+   for (loc_port_num = 1; loc_port_num <= PNET_MAX_PORT; loc_port_num++)
```

Comment:
Use p`f_port_get_next()` instead. See `pf_lldp_init()` for usage example.

Reply:
Done